### PR TITLE
Ranked1v1 refactoring additions

### PR DESCRIPTION
**Pull request by Raistlfiren**
Updated the Ranked1v1 endpoint for the API. I added a new endpoint for `/rating`. If someone accesses `/rating`, then they must specify a `1v1` or `global` argument. This will then get either the rank for 1v1's or global ratings. Added tests for the new endpoint.

### CODE REVIEW

File: api/ranked1v1.py [hunk 1]
```diff
@@ -46,20 +46,82 @@ def ranked1v1():
...
                       args=args, where=where)
 
+@app.route('/rating/<string:rating_type>')
+def rating_type(rating_type):
+    if request.values.get('sort'):
+        raise InvalidUsage('Sorting is not supported for ranked1v1')
+
+    page = int(request.values.get('page[number]', 1))
+    page_size = int(request.values.get('page[size]', MAX_PAGE_SIZE))
+    player = request.args.get('filter[player]')
```
Comment (micheljung) [inline api/ranked1v1.py]: (did we discuss this in slack?) There should not be a filter for player. For single players, there is a separate route

File: api/ranked1v1.py [hunk 2]
```diff
@@ -46,20 +46,82 @@ def ranked1v1():
...
+    result = fetch_data(Ranked1v1Schema(), TABLE1V1, select, MAX_PAGE_SIZE, request,
+                        many=False, where='r.id=%(id)s', args=dict(id=player_id, row_num=0))
+
+    if 'id' not in result['data']:
+        return {'errors': [{'title': 'No entry with this id was found'}]}, 404
+
+    return result
+
+@app.route('/rating/<string:rating_type>/<int:player_id>')
+def rating_type_get_player(rating_type, player_id):
```
Comment (micheljung) [inline api/ranked1v1.py]: Isn't this redundant to the method above?

File: api/ranked1v1.py [hunk 3]
```diff
@@ -46,20 +46,82 @@ def ranked1v1():
...
+        where += " l.login LIKE %(player)s"
+        args['player'] = '%' + player + '%'
+
+    if rating_type == '1v1':
+        table = TABLE1V1
+        select = append_select_expression()
+    elif rating_type == 'global':
+        table = TABLEGLOBAL
+    else:
+        raise InvalidUsage('The rating type should be either 1v1 or global.')
```
Comment (micheljung) [inline api/ranked1v1.py]: Instead of having logic here, I would create two static routes which use a common method

File: api/ranked1v1.py (hunk 1 above)
Comment (Raistlfiren) [inline api/ranked1v1.py]: Haha... This PR was from March 20, a long long time ago. Way before our conversation in slack.

**Comment by IDragonfire** [comment]
can you rebase this commit?
btw. can i remove your branch `pr/26`?

**Comment by Sheeo** [comment]
@Raistlfiren got time for a rebase? :)

**Pull request closed without merging**